x-pack/plugins/lens/public/shared_components/field_picker/field_picker.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import './field_picker.scss';
+import React, { useRef } from 'react';
+import { i18n } from '@kbn/i18n';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
+import { EuiComboBox, EuiComboBoxProps, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import classNames from 'classnames';
+import { DataType } from '../../types';
+import { LensFieldIcon } from './lens_field_icon';
+import { TruncatedLabel } from './truncated_label';
+import type { FieldOptionValue, FieldOption } from './types';
+
+export interface FieldPickerProps<T extends FieldOptionValue>
+  extends EuiComboBoxProps<FieldOption<T>['value']> {
+  options: Array<FieldOption<T>>;
+  selectedField?: string;
+  onChoose: (choice: T | undefined) => void;
+  onDelete?: () => void;
+  fieldIsInvalid: boolean;
+  'data-test-subj'?: string;
+}
+
+const DEFAULT_COMBOBOX_WIDTH = 305;
+const COMBOBOX_PADDINGS = 90;
+const DEFAULT_FONT = '14px Inter';
+
+export function FieldPicker<T extends FieldOptionValue>({
+  selectedOptions,
+  options,
+  onChoose,
+  onDelete,
+  fieldIsInvalid,
+  ['data-test-subj']: dataTestSub,
+  ...rest
+}: FieldPickerProps<T>) {
+  const styledOptions = options?.map(({ compatible, exists, ...otherAttr }) => ({
+    ...otherAttr,
+    compatible,
+    exists,
+    className: classNames({
+      'lnFieldPicker__option--incompatible': !compatible,
+      'lnFieldPicker__option--nonExistant': !exists,
```

Comment:
Are these classes being applied in the appropriate circumstances? When running locally, I noticed that some items in the field selector weren't being given these classes and accompanying color styling that they would have been given previously.

For example, in the screenshot below, I'd expect `@timestamp` and `agent.keyword` fields to be given the incompatible treatment (as the "Median" function is selected).

![image](https://user-images.githubusercontent.com/3884767/175615300-4c75e9e7-a464-469a-8634-48ddb519f654.png)